src/native/main/WebCommonNativeFacade.ts

```diff
@@ -152,4 +178,51 @@ export class WebCommonNativeFacade implements CommonNativeFacade {
 		await locator.initialized
 		return locator
 	}
+
+	private async parseContacts(fileList: FileReference[]) {
+		const { fileApp, logins, fileController } = await WebCommonNativeFacade.getInitializedLocator()
+		const { vCardFileToVCards, vCardListToContacts } = await import("../../contacts/VCardImporter.js")
+
+		await logins.waitForPartialLogin()
+
+		const parsedContacts: Contact[] = []
+		for (const file of fileList) {
+			if (Object.values<string>(VCARD_MIME_TYPES).includes(file.mimeType)) {
+				const dataFile = await fileApp.readDataFile(file.location)
+				if (dataFile == null) continue
+
+				const blob = new Blob([dataFile.data], { type: file.mimeType })
+				const buffer = await blob.arrayBuffer()
+				const decoder = new TextDecoder("utf-8")
+				const parsedVCards = vCardFileToVCards(decoder.decode(buffer))
+				if (parsedVCards == null) continue
+
+				const contactMembership = assertNotNull(logins.getUserController().user.memberships.find((m) => m.groupType === GroupType.Contact))
```

Comment:
We've made a helper on our branch, maybe consider using copying it too?
https://github.com/tutao/tutanota/blob/0a89bb1ff36690d8896deba3544dce64e361c658/src/api/main/UserController.ts#L186-L188

Reply:
Sure